src/app/zap-templates/zcl/data-model/chip/service-area-cluster.xml

```diff
@@ -0,0 +1,243 @@
+      <item value="0x09" name="Around"/>
+      <item value="0x0A" name="On"/>
+      <item value="0x0B" name="Above"/>
+      <item value="0x0C" name="Front Of"/>
+      <item value="0x0D" name="Behind"/>
+    </enum>
+
+
+    <enum name="FloorSurfaceTag" type="enum8">
+      <cluster code="0x0150"/>
+      <item value="0x00" name="Carpet"/>
+      <item value="0x01" name="Ceramic"/>
+      <item value="0x02" name="Concrete"/>
+      <item value="0x03" name="Cork"/>
+      <item value="0x04" name="Deep Carpet"/>
+      <item value="0x05" name="Dirt"/>
+      <item value="0x06" name="Engineered Wood"/>
+      <item value="0x07" name="Glass"/>
+      <item value="0x08" name="Grass"/>
+      <item value="0x09" name="Hardwood"/>
+      <item value="0x0A" name="Laminate"/>
+      <item value="0x0B" name="Linoleum"/>
+      <item value="0x0C" name="Mat"/>
+      <item value="0x0D" name="Metal"/>
+      <item value="0x0E" name="Plastic"/>
+      <item value="0x0F" name="Polished Concrete"/>
+
+      <item value="0x10" name="Rubber"/>
+      <item value="0x11" name="Rug"/>
+      <item value="0x12" name="Sand"/>
+      <item value="0x13" name="Stone"/>
+      <item value="0x14" name="Tatami"/>
+      <item value="0x15" name="Terrazzo"/>
+      <item value="0x16" name="Tile"/>
+      <item value="0x17" name="Vinyl"/>
+    </enum>
+
+
+  <bitmap name="Feature" type="bitmap32">
+    <cluster code="0x0150" />
+    <field name="ListOrder"       mask="0x01" />
+  </bitmap>
+
+  <!-- Enum definitions for structures-->
+  <enum name="OpStatusEnum" type="enum8">
+    <cluster code="0x0150"/>
+    <item value="0x00" name="Pending"/>
+    <item value="0x01" name="Traversing"/>
+    <item value="0x02" name="Operating"/>
+    <item value="0x03" name="Skipped"/>
+    <item value="0x04" name="Completed"/>
+  </enum>
+
+  <!-- Data types -->
+  <struct name="LocationInfoStruct">
+    <cluster code="0x0150"/>
+    <item name="HomeLocationInfo"  type="HomeLocationStruct"    optional="false" isNullable="true"/>
+    <item name="LandmarkTag"       type="LandmarkTag"           optional="false" isNullable="true"/>
+    <item name="PositionTag"       type="PositionTag"           optional="false" isNullable="true"/>
+    <item name="SurfaceTag"        type="FloorSurfaceTag"       optional="false" isNullable="true"/>
+  </struct>
+
+  <struct name="MapStruct">
+    <cluster code="0x0150"/>
+    <item name="MapId"   type="int8u"                optional="false"/>
+    <item name="Name" type="char_string" length="64" optional="false"/>
+  </struct>
+
+  <struct name="LocStruct">
+    <cluster code="0x0150"/>
+    <item name="LocationId"   type="int32u"                  optional="false" isNullable="false"/>
+    <item name="MapId"        type="int8u"                   optional="false" isNullable="true"/>
+    <item name="LocationInfo" type="LocationInfoStruct"      optional="false" isNullable="false"/>
+  </struct>
+
+  <struct name="ProgStruct">
```

Comment:
Should we be a bit more verbose with the name of this structure like `ProgressStruct`?

Reply:
The latest version of the spec PR has it as ProgressStruct.\

Reply:
yes, this is called ProgressStruct in the spec